<a href="https://colab.research.google.com/github/KyleSprague/DCS_210/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
hotels <- read.csv("hotels.csv")

In [2]:
install.packages("ggplot2") 
install.packages("tidyverse") 
install.packages("skimr")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [3]:
library("tidyverse")
library("skimr") 
library("ggplot2")

Warning message in system("timedatectl", intern = TRUE):
“running command 'timedatectl' had status 1”
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6      ✔ purrr   0.3.4 
✔ tibble  3.1.7      ✔ dplyr   1.0.10
✔ tidyr   1.2.1      ✔ stringr 1.4.1 
✔ readr   2.1.2      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


In [4]:
skim(hotels)

── Data Summary ────────────────────────
                           Values
Name                       hotels
Number of rows             119390
Number of columns          32    
_______________________          
Column type frequency:           
  character                14    
  numeric                  18    
________________________         
Group variables            None  

── Variable type: character ────────────────────────────────────────────────────
   skim_variable           n_missing complete_rate min max empty n_unique
 1 hotel                           0             1  10  12     0        2
 2 arrival_date_month              0             1   3   9     0       12
 3 meal                            0             1   2   9     0        5
 4 country                         0             1   2   4     0      178
 5 market_segment                  0             1   6  13     0        8
 6 distribution_channel            0             1   3   9     0        5
 7 reserved_room_ty

Warning message in is.null(text_repr) || nchar(text_repr) == 0L:
“'length(x) = 2 > 1' in coercion to 'logical(1)'”


##ANSWER TO QUESTION 1 

The skimr function prints the number of rows in the data set and the number of columns. The row number can be used to get the total number of bookings for all hotels and the column information tells us the number of variables and also the variable names, which will be useful when we need to select specific columns. We are further given a variable breakdown with the names of the variables and the data type they are associated with (whether character or numeric). This information is useful for manipulating the data if we want to use specific functions that require a particular type of data. In the case of the numeric breakdown, we are also given numeric information such as standard deviation and mean, along with the number of missing data and the completion rate for cells.

In [5]:
##Question 2 code 

hotels %>%
  select(country, lead_time) %>% 
  filter(country != "USA" & lead_time < 1)


country lead_time
1    PRT     0        
2    FRA     0        
3    PRT     0        
4    PRT     0        
5    PRT     0        
6    ESP     0        
7    PRT     0        
8    GBR     0        
9    PRT     0        
10   GBR     0        
11   NLD     0        
12   PRT     0        
13   GBR     0        
14   DEU     0        
15   ESP     0        
16   ESP     0        
17   AUS     0        
18   PRT     0        
19   PRT     0        
20   ESP     0        
21   PRT     0        
22   PRT     0        
23   PRT     0        
24   PRT     0        
25   PRT     0        
26   PRT     0        
27   PRT     0        
28   PRT     0        
29   PRT     0        
30   PRT     0        
⋮    ⋮       ⋮        
6145 PRT     0        
6146 PRT     0        
6147 ITA     0        
6148 FRA     0        
6149 PRT     0        
6150 PRT     0        
6151 DEU     0        
6152 AUS     0        
6153 PRT     0        
6154 DEU     0        
6155 PRT     0        
6156 FRA     0        
6157 DEU     0        
6158 PRT     0        
6159 PRT     0        
6160 PRT     0        
6161 PRT     0        
6162 BRA     0        
6163 DEU     0        
6164 CHN     0        
6165 PRT     0        
6166 GBR     0        
6167 FRA     0        
6168 PRT     0        
6169 PRT     0        
6170 PRT     0        
6171 BEL     0        
6172 NLD     0        
6173 BRA     0        
6174 ROU     0

##ANSWER TO QUESTION 2 

We can use the select function, then the filter function to get data on non-American stays with a lead_time of less than 1 day. 
We see that there are 6174 hotel bookings made on the same day as the hotel stay outside of the US. 

Command:

hotels %>%
  select(country, lead_time) %>% 
  filter(country != "USA" & lead_time < 1)



In [6]:
hotels %>% 
  filter(children >= 1 | babies >= 1)

hotel        is_canceled lead_time arrival_date_year arrival_date_month
1    Resort Hotel 0            18       2015              July              
2    Resort Hotel 1            47       2015              July              
3    Resort Hotel 0             1       2015              July              
4    Resort Hotel 0            10       2015              July              
5    Resort Hotel 1            79       2015              July              
6    Resort Hotel 0           101       2015              July              
7    Resort Hotel 0            92       2015              July              
8    Resort Hotel 1            26       2015              July              
9    Resort Hotel 0           102       2015              July              
10   Resort Hotel 0            78       2015              July              
11   Resort Hotel 0            79       2015              July              
12   Resort Hotel 0             2       2015              July              
13   Resort Hotel 0           109       2015              July              
14   Resort Hotel 1            34       2015              July              
15   Resort Hotel 0            97       2015              July              
16   Resort Hotel 0             8       2015              July              
17   Resort Hotel 0           115       2015              July              
18   Resort Hotel 0            81       2015              July              
19   Resort Hotel 0            84       2015              July              
20   Resort Hotel 0             2       2015              July              
21   Resort Hotel 0            68       2015              July              
22   Resort Hotel 0            21       2015              July              
23   Resort Hotel 0             8       2015              July              
24   Resort Hotel 1           280       2015              July              
25   Resort Hotel 0            17       2015              July              
26   Resort Hotel 0            12       2015              July              
27   Resort Hotel 1            24       2015              July              
28   Resort Hotel 1            44       2015              July              
29   Resort Hotel 0            78       2015              July              
30   Resort Hotel 0            13       2015              July              
⋮    ⋮            ⋮           ⋮         ⋮                 ⋮                 
9303 City Hotel   0            61       2017              August            
9304 City Hotel   0            35       2017              August            
9305 City Hotel   0             8       2017              August            
9306 City Hotel   0           301       2017              August            
9307 City Hotel   0             0       2017              August            
9308 City Hotel   0           221       2017              August            
9309 City Hotel   0            14       2017              August            
9310 City Hotel   0            16       2017              August            
9311 City Hotel   0             0       2017              August            
9312 City Hotel   0           202       2017              August            
9313 City Hotel   0           240       2017              August            
9314 City Hotel   0           130       2017              August            
9315 City Hotel   0           216       2017              August            
9316 City Hotel   0           323       2017              August            
9317 City Hotel   0             5       2017              August            
9318 City Hotel   0             0       2017              August            
9319 City Hotel   0            29       2017              August            
9320 City Hotel   0            22       2017              August            
9321 City Hotel   0           171       2017              August            
9322 City Hotel   0           175       2017              August            
9

##QUESTION 3 

We can use the filter function. Specifically, 

hotels %>% 
  filter(children >= 1 | babies >= 1)

In [7]:
##Code for question 4 
hotels %>% 
  filter(children >= 1 | babies >= 1) %>%
  count(hotel) 


hotel        n   
1 City Hotel   5403
2 Resort Hotel 3929

## ANSWER TO QUESTION 4 

It is more likely that you will find bookings with children or babies in city hotels than in resort hotels. The number for city hotels with at least one child or one baby is 5403 and the number for resort hotels is 3929 bookings

In [8]:
#code for question 5
hotels %>%  
  filter(children > 1 | babies == 1) %>% 
  nrow() 

[1] 4597

In [9]:
#code for question 6 

hotels %>% 
  filter(children > 1 | babies == 1) %>% 
  count(hotel)

hotel        n   
1 City Hotel   2424
2 Resort Hotel 2173

##ANSWERS TO QUESTION 5 and 6

The command I used to find that there are 4597 bookings with more than one child or one baby is: <brk>

hotels %>% 
  filter(children > 1 | babies == 1) %>%
  nrow() <brk>

To get the number of bookings for resort hotels and city hotels, we can use the same command for each. That is, we can use <brk>

hotels %>% 
  filter(children > 1 | babies == 1) %>%
  count(hotel) <brk>

which tells us the number of bookings with more than one child or one baby is 2424 in city hotels and 2173 in resort hotels. 

In [10]:
##CODE FOR QUESTION 7 

##SEE ABOVE 

'''

df <- hotels %>%
  mutate(little_ones = children + babies)    
  arrange(desc(country))


ggplot(df, 
  aes(x = little_ones > 1)) +
  geom_histogram() + 
  facet_wrap(~ hotel)


'''



ERROR: ignored

##ANSWER TO QUESTION 7 

The countries with the highest number of children or babies staying in hotel rooms are in Europe; there are more likely to be children or babies staying at city hotels than at resort hotels as we saw from previous work. 

In [12]:
##CODE FOR QUESTION 8  

hotels %>%   
  count(adults) %>%
  arrange(desc(n)) 


adults n    
1   2     89680
2   1     23027
3   3      6202
4   0       403
5   4        62
6  26         5
7   5         2
8  20         2
9  27         2
10  6         1
11 10         1
12 40         1
13 50         1
14 55         1

##ANSWER TO QUESTION 8 

The most common number of adults in a booking is two at 89680 bookings. The most surprising result is that there were bookings with over twenty people, which will skew the mean

In [13]:
#code for question 9 part 1 

hotels %>%   
  group_by(adults, reservation_status) %>%
  filter(is_canceled == 1) %>%
  count(adults) %>%
  arrange(desc(n)) %>% 
  arrange(desc(adults))



adults reservation_status n    
1  55     Canceled               1
2  50     Canceled               1
3  40     Canceled               1
4  27     Canceled               2
5  26     Canceled               5
6  20     Canceled               2
7  10     Canceled               1
8   6     Canceled               1
9   5     Canceled               2
10  4     Canceled              16
11  3     Canceled            2104
12  3     No-Show               47
13  2     Canceled           34548
14  2     No-Show              710
15  1     Canceled            6228
16  1     No-Show              446
17  0     Canceled             105
18  0     No-Show                4

In [ ]:
#code for question 9 part 2

In [15]:
hotels %>%   
  group_by(adults, reservation_status) %>%
  filter(is_canceled == 0) %>%
  count(adults) %>%
  arrange(desc(n)) %>%  
  arrange(desc(adults)) 



adults reservation_status n    
1 4      Check-Out             46
2 3      Check-Out           4051
3 2      Check-Out          54422
4 1      Check-Out          16353
5 0      Check-Out            294

##QUESTION 9 Answer

Adults are much more likely to complete their stay if they book as a small group of adults rather than as a large group (e.g. 20 adults or more) 









In [16]:
##Code for question 10 

hotels %>%   
  group_by(hotel) %>%
  summarise(
    min_adr = min(adr),
    mean_adr = mean(adr),
    median_adr = median(adr),
    max_adr = max(adr)
    )
  

hotel        min_adr mean_adr  median_adr max_adr
1 City Hotel    0.00   105.30447 99.9       5400   
2 Resort Hotel -6.38    94.95293 75.0        508

##ANSWER TO QUESTION 10
 The hotel type with the higher average adr is the city hotel 105.30 vs 94.95 dollars for resort hotels)

##ANSWER TO QUESTION 11 

The hotel type with the very low minimum is the resort hotel with -6.38 dollars and the hotel with a very high maximum is the city hotel with an adr of $5400. The arrival day of month for the former stay was 5, the week of the year was 10, and the year was 2017. For the max amount, the arrival date was day 25 of week 13 of 2016. In both cases only 2 adults were present. 

In [17]:
min_adr <- 0.00 
max_adr <- 5400 

hotels %>%    
  select(children, adults, babies, hotel, adr, arrival_date_day_of_month, arrival_date_week_number, arrival_date_year) %>%
  filter(adr == 5400) 

hotels %>% 
  select(children, adults, babies, hotel, adr, arrival_date_day_of_month, arrival_date_week_number, arrival_date_year) %>%
  filter(adr == -6.38) 





children adults babies hotel      adr  arrival_date_day_of_month
1 0        2      0      City Hotel 5400 25                       
  arrival_date_week_number arrival_date_year
1 13                       2016

children adults babies hotel        adr   arrival_date_day_of_month
1 0        2      0      Resort Hotel -6.38 5                        
  arrival_date_week_number arrival_date_year
1 10                       2017